Toolkit/AppDeployToolkit/AppDeployToolkitMain.ps1

```diff
@@ -15362,29 +15362,25 @@ https://psappdeploytoolkit.com
     }
     Process {
         Try {
-            $ServiceObject = Get-WmiObject -ComputerName $ComputerName -Class 'Win32_Service' -Filter "Name='$Name'" -ErrorAction 'Stop'
-            # If nothing is returned from Win32_Service, check Win32_BaseService
-            If (-not $ServiceObject) {
-                $ServiceObject = Get-WmiObject -ComputerName $ComputerName -Class 'Win32_BaseService' -Filter "Name='$Name'" -ErrorAction 'Stop'
+            If ($PassThru) {
+                $ServiceObject = Get-WmiObject -ComputerName $ComputerName -Class 'Win32_Service' -Filter "Name='$Name'" -ErrorAction 'Stop'
+                # If nothing is returned from Win32_Service, check Win32_BaseService
+                If (-not $ServiceObject) {
+                    $ServiceObject = Get-WmiObject -ComputerName $ComputerName -Class 'Win32_BaseService' -Filter "Name='$Name'" -ErrorAction 'Stop'
+                }
+            }
+            else {
+                # Known issue that WMI method fails in Sandbox, so only use it if -PassThru requested
+                $ServiceObject = (Get-Service -ComputerName $ComputerName -Name $Name -ErrorAction Ignore).Count -gt 0
```

Comment:
The double bang notation `!!(Get-Service -ComputerName $ComputerName -Name $Name -ErrorAction SilentlyContinue)` is safer here as the automatic `Count` property isn't available on non-array object types when the StrictMode version is 2 or higher.